recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -165,6 +172,24 @@ add_task(withMockExperiments(withMockPreferences(async function (experiments, mo
   startObserver.restore();
 })));
 
+// start should detect if a new preference value type matches the previous value type
+add_task(withMockPreferences(async function (mockPreferences) {
+  mockPreferences.set("fake.type_preference", "oldvalue");
+
+  await Assert.rejects(
+    PreferenceExperiments.start({
```

Comment:
> We should also have a test that the forwarded (wrapAsync) stuff on the NormandyDriver throws, too (so we don't accidentally swallow exceptions there).

That's probably a test for a different PR, in the SandboxManager tests. I'd rather test that once for wrapAsync instead of testing every use of wrapAsync. Although there is the unfilled gap of an integration test for preference-experiment recipes.

> More generally, we should make sure that the start() callers do the right thing with these exceptions. I tried to find them but github search is awful and this all seems to still live on a branch, maybe / sort of? So I couldn't figure out where the entrypoint is...

The entrypoint is in the action code in `recipe-server` that gets run in the sandbox: https://github.com/andymikulski/normandy/blob/d74e71bc7c86d392e193d79a3cc819bb7d2427c6/recipe-server/client/actions/preference-experiment/index.js#L40

Feature branches are... well, they are _a_ solution to a real problem. It's still up for debate if they're a _good_ solution. :P

> In any case, we should make sure that if starting one of these experiments fails, it doesn't break running any/all experiments.

We have existing tests that [ActionSandboxManager throws when the sandbox code throws](https://github.com/andymikulski/normandy/blob/d74e71bc7c86d392e193d79a3cc819bb7d2427c6/recipe-client-addon/test/unit/test_ActionSandboxManager.js#L70). We're missing a test that `RecipeRunner`, which calls that method, handles the error appropriately and doesn't halt execution of other actions. I think that's separate from this PR though, and shouldn't block it.